libraries/AP_Soaring/AP_Soaring.h

```diff
@@ -32,7 +32,7 @@
 class SoaringController {
     Variometer::PolarParams _polarParams;
     ExtendedKalmanFilter _ekf{};
-    AP_TECS &_tecs;
+    class AP_TECS &_tecs;
```

Comment:
new one ? I don't mind getting those but we generally  don't  use them

Reply:
> new one ? I don't mind getting those but we generally don't use them

Using these allows the header to not know about the details of the `AP_TECS` class itself - just that there exists a class called `AP_TECS`.  This only works if you're dealing with a reference or a pointer to the object, as in that case you don't need to know the storage size of the object.